tests/e2e/Services/Functions/FunctionsCustomServerTest.php

```diff
@@ -1803,13 +1804,89 @@ public function testCreateFunctionWithResponseFormatHeader()
             'timeout' => 15,
         ]);
 
-        $this->assertEquals(201, $function['headers']['status-code']);
+        $this->assertEquals(201, $response['headers']['status-code']);
+        $this->assertArrayNotHasKey('scopes', $response['body']);
+        $this->assertArrayNotHasKey('specification', $response['body']);
 
-        $functionId = $function['body']['$id'] ?? '';
+        // get function with 1.5.0 response format header
+        $function = $this->client->call(Client::METHOD_GET, '/functions/' . $response['body']['$id'], array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-response-format' => '1.5.0', // add response format header
+        ], $this->getHeaders()));
+
+        $this->assertEquals(200, $function['headers']['status-code']);
+        $this->assertArrayNotHasKey('scopes', $function['body']);
+        $this->assertArrayNotHasKey('specification', $function['body']);
+
+        // get function without response format header
+        $function = $this->client->call(Client::METHOD_GET, '/functions/' . $response['body']['$id'], array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()));
 
+        $this->assertEquals(200, $function['headers']['status-code']);
+        $this->assertArrayHasKey('scopes', $function['body']);
+        $this->assertArrayHasKey('specification', $function['body']);
+
+        $functionId = $function['body']['$id'] ?? '';
         $this->cleanupFunction($functionId);
     }
 
+    public function testRequestFilters()
+    {
+        // create function
+        $response = $this->client->call(Client::METHOD_POST, '/functions', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'functionId' => ID::unique(),
+            'name' => 'Test',
+            'runtime' => 'php-8.0',
+            'entrypoint' => 'index.php',
+            'timeout' => 15,
+        ]);
+
+        $this->assertEquals(201, $response['headers']['status-code']);
+        $function1Id = $response['body']['$id'];
+
+        // create another function
+        $response = $this->client->call(Client::METHOD_POST, '/functions', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'functionId' => ID::unique(),
+            'name' => 'Test2',
+            'runtime' => 'php-8.0',
+            'entrypoint' => 'index.php',
+            'timeout' => 15,
+        ]);
```

Comment:
Let's use `$this->setupFunction()` helper in those 2 calls instead. It also returns functionId, which is exactly what we need.